modules/miner.go

```diff
@@ -14,6 +15,15 @@ type MinerInfo struct {
 	Address        types.UnlockHash
 }
 
+// MinerWork is all the information a miner needs to do work
+type MinerWork struct {
+	Block      types.Block
+	ParentID   types.BlockID
+	Nonce      uint64
+	MerkleRoot crypto.Hash
+	Target     types.Target
+}
```

Comment:
oh yeah, you need the target.

but you don't need to assemble the block from the fpga, you can just write a 'submitheader' api call. I guess that could get complicated though if new transactions are coming in at a rapid rate.